tempesta_fw/ss_skb.c

```diff
@@ -247,15 +247,17 @@ __skb_data_address(struct sk_buff *skb)
  * fragment is not there, then find the next data location.
  */
 static inline void
-__it_next_data(struct sk_buff *skb, int i, TfwStr *it)
+__it_next_data(struct sk_buff *skb, int i, TfwStr *it, int *fragn)
 {
 	struct skb_shared_info *si = skb_shinfo(skb);
 	if (i < si->nr_frags) {
 		it->data = skb_frag_address(&si->frags[i]);
 		it->skb = skb;
+		*fragn = i;
 	} else {
 		it->skb = skb->next;
 		it->data = __skb_data_address(it->skb);
+		*fragn = -1;
```

Comment:
`__skb_data_address()` can return either a pointer to skb's linear part, or it can return a pointer to the first fragment. `*fragn` may be `0` sometimes.

Reply:
The fix looks good.